expression/chunk_executor.go

```diff
@@ -391,3 +403,16 @@ func VectorizedFilterConsiderNull(ctx sessionctx.Context, filters []Expression,
 	}
 	return selected, isNull, nil
 }
+
+// VectorizedFilterConsiderNull2 applies a list of filters to a Chunk and
+// returns two bool slices, `selected` indicates whether a row passed the
+// filters, `isNull` indicates whether the result of the filter is null.
+// Filters is executed vectorized by chunk
+func VectorizedFilterConsiderNull2(ctx sessionctx.Context, filters []Expression, iterator *chunk.Iterator4Chunk, selected []bool, isNull []bool) ([]bool, []bool, error) {
+	selected, isNull, err := VecEvalBool(ctx, filters, iterator.GetChunk(), selected, isNull)
```

Comment:
It seems `VecEvalBool` ignores `iterator.GetChunk().sel` here.

Reply:
The ```selected``` slice is initialized in the ```VecEvalBool``` function. Both ```vectorizedFilterConsiderNull2``` and ```VectorizedFilterConsiderNull``` will initialize the ```selected``` slice. So we don't care ```iterator.GetChunk().sel```.